docs/Examples.md

```diff
@@ -3,72 +3,100 @@ Examples
 
 So how do you use this library? Well, it's pretty easy. Just follow this
 template. First, set up an `enum` with all of your API targets. Note that you
-can include information as part of your enum. Let's look at a simple example.
+can include information as part of your enum. Let's look at a common example. First we create a new file named `MyService.swift`:
 
 ```swift
-enum GitHub {
+enum MyService {
     case Zen
-    case UserProfile(String)
+    case ShowUser(id: Int)
+    case UpdateUser(id: Int, firstName: String, lastName: String)
 }
 ```
 
 This enum is used to make sure that you provide implementation details for each
-target (at compile time). The enum *must* conform to the `TargetType` protocol.
-Let's take a look at what that might look like.
+target (at compile time). You can see that parameters needed for requests can be defined as per the enum cases parameters. The enum *must* additionally conform to the `TargetType` protocol. Let's get this done via an extension in the same file:
 
 ```swift
-private extension String {
-    var URLEscapedString: String {
-        return self.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.URLHostAllowedCharacterSet())!
-    }
-}
-
-extension GitHub: TargetType {
-    var baseURL: NSURL { return NSURL(string: "https://api.github.com")! }
+// MARK: - TargetType Protocol Implementation
+extension MyService: TargetType {
+    var baseURL: NSURL { return NSURL(string: "https://api.myservice.com")! }
     var path: String {
         switch self {
         case .Zen:
             return "/zen"
-        case .UserProfile(let name):
-            return "/users/\(name.URLEscapedString)"
+        case .ShowUser(let id):
+            return "/users/\(id)"
+        case .UpdateUser(let id, _, _):
+            return "/users/\(id)"
         }
     }
     var method: Moya.Method {
-        // all requests in this example will use GET.  Usually you would switch
-        // on the enum, like we did in `var path: String`
-        return .GET
+        switch self {
+        case .Zen, .ShowUser(_):
+            return .GET
+        case .UpdateUser(_, _, _):
```

Comment:
Cool, you're right, I was using it like that already but thought that might be a Swift side effect and actually undocumented. But now you say it, I can clean it up. Thanks!